gobblin-compaction/src/main/java/gobblin/compaction/source/CompactionSource.java

```diff
@@ -96,4 +104,24 @@ protected FileSystem getSourceFileSystem(State state)
     return HadoopUtils.getOptionallyThrottledFileSystem(FileSystem.get(URI.create(uri), conf), state);
   }
 
+  private void copyJarDependencies(FileSystem fs, State state) throws IOException {
+
+    if (!state.contains(ConfigurationKeys.JOB_JAR_FILES_KEY)) {
+      return;
+    }
+
+    this.tmpOutputDir = state.getProp(MRCompactor.COMPACTION_TMP_DEST_DIR, MRCompactor.DEFAULT_COMPACTION_TMP_DEST_DIR);
+    LocalFileSystem lfs = FileSystem.getLocal(HadoopUtils.getConfFromState(state));
+    Path tmpJarFileDir = new Path(this.tmpOutputDir, "_gobblin_compaction_jars");
+    state.setProp (MRCompactor.COMPACTION_JARS, tmpJarFileDir.toString());
+
+    fs.delete(tmpJarFileDir, true);
```

Comment:
If there are multiple compaction jobs running, this could make them interfere with each other.